Source/WebKit/UIProcess/ProvisionalPageProxy.cpp

```diff
@@ -253,29 +253,41 @@ void ProvisionalPageProxy::initializeWebPage(RefPtr<API::WebsitePolicies>&& webs
     if (websitePolicies)
         m_mainFrameWebsitePoliciesData = makeUnique<WebsitePoliciesData>(websitePolicies->data());
 
-    Ref protectedProcess = this->protectedProcess();
+    Ref process = this->process();
     if (page->preferences().siteIsolationEnabled()) {
-        RegistrableDomain navigationDomain(m_request.url());
-        if (auto existingRemotePageProxy = m_browsingContextGroup->takeRemotePageInProcessForProvisionalPage(page, protectedProcess)) {
+        if (RefPtr existingRemotePageProxy = m_browsingContextGroup->takeRemotePageInProcessForProvisionalPage(page, process)) {
             m_webPageID = existingRemotePageProxy->pageID();
             m_mainFrame = existingRemotePageProxy->page()->mainFrame();
             m_needsMainFrameObserver = false;
             m_messageReceiverRegistration.stopReceivingMessages();
             m_messageReceiverRegistration.transferMessageReceivingFrom(existingRemotePageProxy->messageReceiverRegistration(), *this);
-            send(Messages::WebPage::CreateProvisionalFrame(ProvisionalFrameCreationParameters {
-                std::nullopt,
-                m_mainFrame->effectiveSandboxFlags(),
-                m_mainFrame->scrollingMode(),
-            }, m_mainFrame->frameID()));
+            m_needsDidStartProvisionalLoad = false;
             m_needsCookieAccessAddedInNetworkProcess = true;
             registerWithInspectorController = false; // FIXME: <rdar://121240770> This is a hack. There seems to be a bug in our interaction with WebPageInspectorController.
+        } else {
+            m_browsingContextGroup->addFrameProcessAndInjectPageContextIf(m_frameProcess, [this](WebPageProxy& page) {
```

Comment:
Can we capture m_page instead of this?  That would make it more clear we are not unsafely using pointers.